File: golang/codec/message_open.go
package codec

import (
	"encoding/binary"
	"fmt"
)

type OpenMessage struct {
	SenderID      uint32
	WindowSize    uint32
	MaxPacketSize uint32
}

func (msg OpenMessage) String() string {
	return fmt.Sprintf("{OpenMessage SenderID:%d WindowSize:%d MaxPacketSize:%d}",
		msg.SenderID, msg.WindowSize, msg.MaxPacketSize)
}

func (msg OpenMessage) Channel() (uint32, bool) {
	return 0, false
}

func (msg OpenMessage) MarshalMux() ([]byte, error) {
	packet := make([]byte, payloadSizes[msgChannelOpen]+1)
	packet[0] = msgChannelOpen
	binary.BigEndian.PutUint32(packet[1:5], msg.SenderID)
	binary.BigEndian.PutUint32(packet[5:9], msg.WindowSize)
	binary.BigEndian.PutUint32(packet[9:13], msg.MaxPacketSize)
	return packet, nil
}

func (msg *OpenMessage) UnmarshalMux(b []byte) error {
	msg.SenderID = binary.BigEndian.Uint32(b[1:5])
	msg.WindowSize = binary.BigEndian.Uint32(b[5:9])
	msg.MaxPacketSize = binary.BigEndian.Uint32(b[9:13])
	return nil
}

File: csharp/qmux/codec/message_open.cs
namespace qmux.codec;

using System.IO;
using System.Text;

public struct OpenMessage : IMessage
{
    private nuint senderId;
    private nuint windowSize;
    private nuint maxPacketSize;
    public string String()
    {
        return $"OpenMessage SenderId:{this.senderId} WindowSize:{this.windowSize} MaxPacketSize:{this.maxPacketSize}";
    }
    public (nuint, bool) Channel()
    {
        return (0, false);
    }
    public byte[] MarshalMux()
    {
        // 12 + 1
        var bufferSize = (int)PayloadSizes.MessageChannelOpen + 1;
        var stream = new MemoryStream(new byte[bufferSize]);
        using (var writer = new BinaryWriter(stream, Encoding.BigEndianUnicode, false))
        {
            // 0
            writer.Write((byte)MessageType.MessageChannelOpen);
            // 1-5
            writer.Write(this.senderId);
            // 5-9
            writer.Write(this.windowSize);
            // 10-14
            writer.Write(this.maxPacketSize);
        }
        return stream.ToArray();
    }
    public void UnmarshalMux(byte[] b)
    {
        var stream = new MemoryStream(b);
        using (var reader = new BinaryReader(stream, Encoding.BigEndianUnicode))
        {
            // 0
            reader.ReadByte();
            // 1-5
            this.senderId = reader.ReadUInt32();
            // 5-9
            this.windowSize = reader.ReadUInt32();
            // 10-14
            this.maxPacketSize = reader.ReadUInt32();
        }
    }
}